pkg/metrics/metrics.go

```diff
@@ -0,0 +1,98 @@
+package metrics
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Error to be  returned when marketplace is unable to reach quay.io.
+type OpsrcError string
+
+const (
+	// Default values which are to be used as labels
+	// for prometheus metrics.
+	Error     = "opsrc_error"
+	OpsrcName = "opsrc_name"
+
+	// Path, Port and Host where custom metrics would be exposed.
+	MetricsPath = "/metrics"
+	MetricsPort = "8383"
+	MetricsHost = "0.0.0.0"
+
+	// Error message to be returned when quay.io is down.
+	UnreachableError OpsrcError = "quay_unavailabe"
+
+	// Error message to be returned when host address is wrong.
+	NoHostExsistsError OpsrcError = "no_host_exsists"
+)
+
+// Metric variables that report error based on the operator-source
+// which fails to reconcile.
+var (
+	operatorSourceError = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "operator_source_failure_count",
```

Comment:
This could be `opsrc_failures`, unless promethes has some convention I am unaware of. Thoughts @kevinrizza ?

Reply:
`opsrc_errors` works here as well.

Reply:
Prometheus requirements are just to have "_" between words. Since this is of the type `counter` in Prometheus metrics, I named it as `operator_source_failure_count`. This could be anything though..

Reply:
That's good to know, let's see what name @kevinrizza thinks is good, perhaps `opsrc_failure_count`?

Reply:
I think keeping our variable names short, especially when they are going to be displayed in a UI is good. `opsrc_failure_count` seems fine.